androidApp/src/main/java/com/mbta/tid/mbta_app/android/map/MapViewModel.kt

```diff
@@ -94,12 +102,20 @@ open class MapViewModel(
         _config.value = latestConfig
     }
 
-    override fun globalMapData(now: Instant): GlobalMapData? =
-        runBlocking(Dispatchers.IO) {
-            globalResponse.first()?.let {
-                GlobalMapData(it, GlobalMapData.getAlertsByStop(it, alertsData, now))
-            }
+    override suspend fun globalMapData(now: Instant): GlobalMapData? =
+        globalResponse.first()?.let {
+            GlobalMapData(it, GlobalMapData.getAlertsByStop(it, alertsData, now))
+        }
+
+    @Composable
+    override fun rememberGlobalMapData(now: Instant): GlobalMapData? {
+        var result by remember { mutableStateOf<GlobalMapData?>(null) }
```

Comment:
suggestion(non-blocking): For consistency with the `refresh` functions below, this could update a mutable state property of the VM rather than remembering the result itself.

Reply:
I don't think it works the same way as those `refresh` functions - it's a `@Composable` called from the UI thread and not a `suspend fun` called from a `LaunchedEffect`, and even if I made `globalMapData()` update some mutable state it'd still need to return the value to avoid uglier code or less up-to-date output of the `refresh` functions.